usl_pipeline/usl_lib/usl_lib/chunkers/polygon_chunkers.py

```diff
@@ -0,0 +1,136 @@
+import pathlib
+from typing import Iterable, Tuple
+
+from shapely import geometry
+
+from usl_lib.chunkers import chunkers_data
+from usl_lib.shared import geo_data
+from usl_lib.writers import polygon_writers
+
+
+def _get_chunk_bounding_box(
+    elevation_header: geo_data.ElevationHeader,
+    chunk_size: int,
+    y_chunk_index: int,
+    x_chunk_index: int,
+) -> Tuple[float, float, float, float]:
+    """Calculates bounding box in coordinate values corresponding to a chunk region.
+
+    Args:
+        elevation_header: Information about cell grid projection to coordinate system.
+        chunk_size: Size of chunk in cells.
+        y_chunk_index: Index of the chunk along the Y-axis
+        x_chunk_index: Index of the chunk along the X-axis
+
+    Returns:
+        The bounding box of the chunk represented by a tuple (min-x, min-y, max-x,
+        max-y).
+    """
+    global_min_x = elevation_header.x_ll_corner
+    cell_size = elevation_header.cell_size
+    global_col_count = elevation_header.col_count
+    global_row_count = elevation_header.row_count
+    global_max_y = elevation_header.y_ll_corner + cell_size * global_row_count
+
+    # X-axis goes from left to right (from smaller cell index to greater one).
+    chunk_min_x = global_min_x + cell_size * x_chunk_index * chunk_size
+    # Y-axis goes from bottom up (from greater cell index to smaller one).
+    chunk_max_y = global_max_y - cell_size * y_chunk_index * chunk_size
+
+    chunk_col_count = min(chunk_size, global_col_count - x_chunk_index * chunk_size)
+    chunk_row_count = min(chunk_size, global_row_count - y_chunk_index * chunk_size)
+    chunk_max_x = chunk_min_x + chunk_col_count * cell_size
+    chunk_min_y = chunk_max_y - chunk_row_count * cell_size
+    return chunk_min_x, chunk_min_y, chunk_max_x, chunk_max_y
+
+
+def _bbox_intersection(
+    bbox1: Tuple[float, float, float, float], bbox2: Tuple[float, float, float, float]
```

Comment:
If you get tired of typing `Tuple[float, float, float, float]` a bunch, you can do something like
```python
BoundingBox = Tuple[float, float, float, float]

def _bbox_intersection(bbox1: BoundingBox, bbox2: BoundingBox) -> bool:
```

Reply:
Perfect!